src/poll.rs

```diff
@@ -183,17 +185,16 @@ impl ZmqPoller {
         if events.contains(event) {
             Poll::Ready(Ok(()))
         } else {
-            self.clear_read_ready(cx, Ready::readable())?;
+            self.clear_read_ready(cx)?;
             Poll::Pending
         }
     }
-}
 
-impl Deref for ZmqPoller {
-    type Target = PollEvented<SocketWrapper>;
-
-    fn deref(&self) -> &Self::Target {
-        &self.0
+    fn clear_read_ready(&self, cx: &mut Context<'_>) -> Result<()> {
```

Comment:
Nice job working around the guard API!  It sure is awkward for this particular use case.